File: src/main/java/chapter5/schedulers/IOSchedulerExample.java
package chapter5.schedulers;

import common.CommonUtils;
import common.Log;
import io.reactivex.Observable;
import io.reactivex.schedulers.Schedulers;

import java.io.File;


public class IOSchedulerExample {

    public void basic() {
        String root = "/Users/n2soft/Desktop";
        File[] files = new File(root).listFiles();
        Observable<String> source = Observable.fromArray(files)
                .filter(f -> !f.isDirectory())
                .map(f -> f.getAbsolutePath())
                .subscribeOn(Schedulers.io());

        source.subscribe(Log::it);
        CommonUtils.sleep(500);
    }

    public static void main(String[] args) {
        IOSchedulerExample demo = new IOSchedulerExample();
        demo.basic();
    }

}

File: src/main/kotlin/chapter5/schedulers/IOSchedulerExample_.kt
package chapter5.schedulers

import common.CommonUtils_
import common.Log_
import io.reactivex.Observable
import io.reactivex.schedulers.Schedulers
import java.io.File


class IOSchedulerExample_ {

    fun basic() {
        val root = "/Users/n2soft/Desktop"
        val files = File(root).listFiles()
        val source = Observable.fromArray(*files!!)
                .filter { !it.isDirectory }
                .map { it.absolutePath }
                .subscribeOn(Schedulers.io())

        source.subscribe { Log_.it(it) }
        CommonUtils_.sleep(500)
    }

}


fun main(args: Array<String>) {
    val demo = IOSchedulerExample_()
    demo.basic()
}
